atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -50,25 +53,188 @@ def __init__(self, tasker, workflow, replacement_pullspecs=None):
         self.replacement_pullspecs = replacement_pullspecs
 
     def run(self):
+        """
+        Run pin_operator_digest plugin
+        """
         if self.should_run():
+            # TODO: check FROM scratch - here? check_user_settings?
             if self.is_in_orchestrator():
                 return self.run_in_orchestrator()
             else:
                 return self.run_in_worker()
 
     def should_run(self):
+        """
+        Determine if this is an operator manifest bundle build
+
+        :return: bool, should plugin run?
+        """
         if has_operator_bundle_manifest(self.workflow):
             return True
         else:
             self.log.info("Not an operator manifest bundle build, skipping plugin")
             return False
 
     def run_in_orchestrator(self):
+        """
+        Run plugin in orchestrator. Find all image pullspecs,
+        compute their replacements and set build arg for worker
+        """
         try:
             self.site_config = get_operator_manifests(self.workflow)
         except KeyError:
             raise RuntimeError("operator_manifests configuration missing in reactor config map")
-        override_build_kwarg(self.workflow, "operator_bundle_replacement_pullspecs", {})
+
+        operator_manifest = self._get_operator_manifest()
+        pullspecs = self._get_pullspecs(operator_manifest)
+
+        if pullspecs:
+            replacement_pullspecs = self._get_replacement_pullspecs(pullspecs)
+            self._set_worker_arg(replacement_pullspecs)
 
     def run_in_worker(self):
-        pass
+        """
+        Run plugin in worker. Replace image pullspecs
+        based on replacements computed in orchestrator.
+        """
+        if not self.replacement_pullspecs:
+            self.log.info("No pullspecs need to be replaced")
+            return
+
+        operator_manifest = self._get_operator_manifest()
+        replacement_pullspecs = {
+            ImageName.parse(old): ImageName.parse(new)
+            for old, new in self.replacement_pullspecs.items()
+        }
+
+        for operator_csv in operator_manifest.files:
+            operator_csv.replace_pullspecs(replacement_pullspecs)
+            operator_csv.dump()
+            self.log.info("Replaced pullspecs in %s", operator_csv.path)
+
+    def _get_operator_manifest(self):
+        if self.user_config is None:
+            raise RuntimeError("operator_manifest configuration missing in container.yaml")
+
+        manifests_dir = os.path.join(self.workflow.source.path, self.user_config["manifests_dir"])
+        self.log.info("Looking for operator CSV files in %s", manifests_dir)
+
+        operator_manifest = OperatorManifest.from_directory(manifests_dir)
+
+        if operator_manifest.files:
+            # TODO: log paths in a particular order? make sure OperatorManifest orders them?
```

Comment:
Does order matter? If you want to make them easy to read, feel free to sort it.

Reply:
It was mostly just to make log output deterministic, but I got around that in unit tests so it's probably fine